dask_expr/tests/test_collection.py

```diff
@@ -1372,6 +1372,16 @@ def test_columns_assignment():
     assert_eq(df, ddf)
 
 
+def test_parent_mixed_column_assignment():
+    pdf = pd.DataFrame(list(range(100)))
+    df = from_pandas(pdf, npartitions=4)
+    df["fold"] = 0
+    df["fold"] = df["fold"].map_partitions(
+        lambda x: pd.Series(np.random.randint(0, 4, len(x)))
+    )
+    df.simplify()
```

Comment:
It does seem like the logic needed to reproduce the failure is a bit weird. I'm open to other ideas :)

Reply:
This potentially converts from a series to a df if you do the conversion
Unconditionally, that needs a bit of care before merging

Reply:
Are you saying there is an issue with the test or with the `tolist()` addition? (sorry for being slow)

Reply:
Sorry I kind of messed this up in my head, your solution is good